common/src/main/scala/uk/ac/wellcome/platform/archive/common/bagit/models/BagManifest.scala

```diff
@@ -0,0 +1,51 @@
+package uk.ac.wellcome.platform.archive.common.bagit.models
+
+import java.io.InputStream
+
+import uk.ac.wellcome.platform.archive.common.verify.{ChecksumValue, HashingAlgorithm}
+
+import scala.util.matching.Regex
+import scala.util.{Failure, Success, Try}
+
+case class BagManifest(
+                        checksumAlgorithm: HashingAlgorithm,
+                        files: List[BagFile]
+                      )
+
+object BagManifest {
+  val lineRegex: Regex = """(.+?)\s+(.+)""".r
+
+  def create(stream: InputStream, algorithm: HashingAlgorithm): Try[BagManifest] = {
+
+    val lines = scala.io.Source
+      .fromInputStream(stream)
+      .mkString
+      .split("\n")
+      .filter { _.nonEmpty }
+      .toList
+
+    val eitherFiles = lines.zipWithIndex.map {
+      case (l, i)  => createBagFile(i, l, None)
```

Comment:
These line numbers won’t necessarily match the line numbers in the file, because you threw away all the empty lines.

Reply:
Also, named parameters and better variable names? It's very non-obvious what `None` is in this scenario.

Reply:
Going to just dump this as we don't actually use it!